src/main/java/io/jenkins/plugins/coverage/source/SourceFileResolver.java

```diff
@@ -0,0 +1,192 @@
+package io.jenkins.plugins.coverage.source;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.FilePath;
+import hudson.model.Describable;
+import hudson.model.Descriptor;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.remoting.VirtualChannel;
+import hudson.util.ListBoxModel;
+import io.jenkins.plugins.coverage.BuildUtils;
+import io.jenkins.plugins.coverage.exception.CoverageException;
+import io.jenkins.plugins.coverage.targets.CoveragePaint;
+import jenkins.MasterToSlaveFileCallable;
+import jenkins.model.Jenkins;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.util.Map;
+
+public class SourceFileResolver implements ExtensionPoint, Describable<SourceFileResolver> {
```

Comment:
Seems the implementation is heavily integrated into the extension point. I would suggest splitting it to the implementation and abstract class